labrad/types.py

```diff
@@ -394,6 +389,28 @@ def _flatten_to(obj, types, endianness):
     return s, t
 
 
+def __flatten__(t, obj, endianness):
+    """Flatten an object to the specific given type.
```

Comment:
I agree on all fronts. I've added a method `flatten` on `LRType` which handles the case where the data is already flattened, and otherwise delegates real flattening work to the existing `__flatten__` method (which we might want to rename).